lib/spack/spack/build_systems/oneapi.py

```diff
@@ -64,6 +79,39 @@ def _oneapi_file(self, version, release):
             self._url_name, version, release['build'])
 
 
+class IntelOneApiCompilerPackage(IntelOneApiPackage):
+    """Base class for Intel oneAPI compiler packages."""
+
+    depends_on('patchelf', type='build')
+
+    def _join_prefix(self, path):
+        return join(self.prefix, 'compiler/latest/linux', path)
+
+    def _ld_library_path(self):
+        dirs = ['lib',
+                'lib/x64',
+                'lib/emu',
+                'lib/oclfpga/host/linux64/lib',
+                'lib/oclfpga/linux64/lib',
+                'compiler/lib/intel64_lin',
+                'compiler/lib']
+        for dir in dirs:
+            yield self._join_prefix(dir)
+
+    def install(self, spec, prefix):
+        super(IntelOneApiCompilerPackage, self).install(spec, prefix)
+
+        rpath = ':'.join(self._ld_library_path())
+        patch_dirs = ['compiler/lib/intel64_lin',
+                      'compiler/lib/intel64',
+                      'bin']
+        for pd in patch_dirs:
+            for file in glob.glob(self._join_prefix(join(pd, '*'))):
+                # Try to patch all files, patchelf will do nothing if
+                # file should not be patched
+                subprocess.call(['patchelf', '--set-rpath', rpath, file])
```

Comment:
Why are the compiler binaries being patched?

Reply:
This was requested by @scheibelp to make find work. icx/ifx cannot be invoked unless you set LD_LIBRARY_PATH, and find needs to invoke the compiler to check the version string. Patching RPATH eliminates the need to set LD_LIBRARY_PATH.

Reply:
Ah, interesting. Thanks.